exts/mod.py

```diff
@@ -0,0 +1,363 @@
+            "The reason behind why you want to add slowmode.", default="N/A"
+        ),
+    ):
+        slowmode_channel: naff.GuildText = channel or ctx.channel
+
+        # working around https://github.com/NAFTeam/NAFF/pull/738 - don't mind me
+        base_channel = naff.BaseChannel.from_dict({"id": slowmode_channel.id, "type": 0}, self.bot)
+        await base_channel.edit(rate_limit_per_user=time, reason=reason)
+        await ctx.send(
+            f":white_check_mark: {slowmode_channel.mention}'s slowmode was set.", ephemeral=True
+        )
+
+    @channel.subcommand(sub_cmd_name="lock", sub_cmd_description="Locks a channel.")
+    async def lock(
+        self,
+        ctx: naff.InteractionContext,
+        channel: typing.Optional[naff.GuildText] = tansy.Option(
+            "The channel that should be locked.", default=None
+        ),
+        reason: str = tansy.Option(
+            "The reason behind why you want to lock the channel.", default="N/A"
+        ),
+    ):
+        lock_channel: naff.GuildText = channel or ctx.channel
+
+        overwrites = lock_channel.permission_overwrites
+
+        changed_guild_overwrite = False
+
+        for overwrite in overwrites:
+            if overwrite.id != METADATA["roles"]["Moderator"]:
+                overwrite.add_denies(naff.Permissions.SEND_MESSAGES, naff.Permissions.ADD_REACTIONS)
+
+                if overwrite.id == ctx.guild.id:
+                    changed_guild_overwrite = True
+
+        if not changed_guild_overwrite:
+            overwrites.append(
+                naff.PermissionOverwrite(
+                    id=ctx.guild.id,
+                    type=naff.OverwriteTypes.ROLE,
+                    deny=naff.Permissions.SEND_MESSAGES | naff.Permissions.ADD_REACTIONS,
+                )
+            )
+
+        await lock_channel.edit(permission_overwrites=overwrites, reason=reason)
+        await ctx.send(f":white_check_mark: {lock_channel.mention} was locked. ", ephemeral=True)
+
+    @channel.subcommand(sub_cmd_name="unlock", sub_cmd_description="Unlocks a channel.")
+    async def unlock(
+        self,
+        ctx: naff.InteractionContext,
+        channel: typing.Optional[naff.GuildText] = tansy.Option(
+            "The channel that should be unlocked.", default=None
+        ),
+        reason: str = tansy.Option(
+            "The reason behind why you want to unlock the channel.", default="N/A"
+        ),
+    ):
+        unlock_channel: naff.GuildText = channel or ctx.channel
+
+        overwrites = unlock_channel.permission_overwrites
+        for overwrite in overwrites:
+            if overwrite.id != METADATA["roles"]["Moderator"]:
+                overwrite.add_allows(naff.Permissions.SEND_MESSAGES, naff.Permissions.ADD_REACTIONS)
+
+        await unlock_channel.edit(permission_overwrites=overwrites, reason=reason)
+        await ctx.send(f":white_check_mark: {unlock_channel.mention} was unlocked.", ephemeral=True)
+
+    @naff.prefixed_command()
```

Comment:
Are we running a perms check on this?

Reply:
We are - there's an extension-wide check that's being used here.